### ISSUE TITLE
Could not find any USB device

### ISSUE THREAD

**Issue by oaksoftYH** [Genymobile/scrcpy#3289]
When using the --otg parameter, I get this error:

scrcpy 1.24 <https://github.com/Genymobile/scrcpy>
INFO: Killing adb daemon (if any)...
ERROR: Could not find any USB device

**Comment by rom1v** [comment]
On Windows I guess? Is USB debugging enabled? (#3050)

**Comment by oaksoftYH** [comment]
Yes, I'm having this error on Windows. My USB debugging is active

**Comment by rom1v** [comment]
Does scrcpy work for mirroring (without the otg option)?

What is the result of:
```
scrcpy --otg -Vdebug
```

Also see #3052 comments.

**Comment by oaksoftYH** [comment]
![image](https://user-images.githubusercontent.com/66079964/169899312-9efc2457-7051-4066-83e9-4802e99a8fdd.png)

Here is the result of scrcpy --otg -Vdebug

**Comment by oaksoftYH** [comment]
??

**Comment by Kamix-GLX** [comment]
i have the same error, i searched on issues and FAQ for a while but i didnt found any solutions :(

**Comment by ByteBreacher** [comment]
### **THIS WILL REALLY HELP YOU** (almost 90% of the time this might be the problem)
  
    Download the latest version of ADB from the official Android developer website.

    Extract the contents of the ADB ZIP file to a directory on your system, such as ~/Tools.

    Add the directory containing the ADB binary to your PATH environment variable by adding the following line to your .bashrc file:
                 
            'export PATH=$PATH:/path/to/adb/'

Reload the .bashrc file by running the following command:
           '. ~/.bashrc'


Try running scrcpy again and see if the error message goes away.

**Comment by FBIGunner** [comment]
> **THIS WILL REALLY HELP YOU** (almost 90% of the time this might be the problem)
> 
> 1. Download the latest version of ADB from the official Android developer website:
>    [https://developer.android.com/studio/releases/platform-tools]
> 2. Extract the contents of the ADB ZIP file to a directory on your system, such as ~/Tools.
> 3. Add the directory containing the ADB binary to your PATH environment variable by adding the following line to your .bashrc file:
>    ```
>      `export PATH=$PATH:/path/to/adb-file-just-downloaded-and-extracted/`
>    ```
> 4. Reload the .bashrc file by running the following command:
>    `. ~/.bashrc`
> 5. Try running `scrcpy` again and see if the error message goes away.

How to do this in Windows? As i searched this is for Linux not for the Windows is there any solution for Windows? Thanks

**Comment by FBIGunner** [comment]
Okay i think that i have a work around for this issue
Just change usb mode on ur phone into charge mode and not MTP mode and it will work perfectly (as i tested lol)

**Comment by rom1v** [comment]
Please also read https://github.com/Genymobile/scrcpy/blob/master/FAQ.md#hidotg-issues-on-windows

**Comment by FBIGunner** [comment]
I tried to reinstall the driver but no luck, the phone i use is LG V60(Android 12) and it has auto mode when you connect to usb it change to mode MTP(file transfer) and when i tried to change it into charge only mode it work as it should
I don't think that this method is 90% sure will fix the problem but it is worth a try!

**Comment by kaptain756464** [comment]
![Screenshot (42)](https://github.com/Genymobile/scrcpy/assets/126095024/374989f2-70df-4f60-a9d6-910a1305a05f)

My Phone is Already connected right now as transfer files and nothing appear .. so what i can do ?

also i try another devices same problem otherwise my cable is the original one

**Comment by FBIGunner** [comment]
Did you installed adb and connect it through adb as u should?, you should try to change usb mode on your phone to charging and try again, if not install your phone driver

**Comment by brunoman4006** [comment]
I have the same issue, but when using -Vdebug it throws this error:

scrcpy 3.1 <https://github.com/Genymobile/scrcpy>
INFO: Killing adb server (if any)...
DEBUG: Open USB device 04e7:0022: libusb error: Input/Output Error
DEBUG: Open USB device 2207:0011: libusb error: Entity not found
ERROR: Could not find any USB device

**Comment by MuslimHS** [comment]
> On my Windows PC, after using "Zadig" to change the usb driver of my device from the default driver to the "WinUSB" driver, I was able to use: 'scrcpy --keyboard=aoa --otg -Vdebug'
> 
> the mouse and keyboard worked on a smashed xperia tablet with minimal issues (sometimes random disconnection, just ran the command again) i was then able to enable developer options and in turn adb debugging.

Worked like a charm!

**Xiaomi Redmi Note 4 and Win10**

Thank you!
